crates/store/re_types_core/src/tuid.rs

```diff
@@ -45,25 +45,25 @@ impl Loggable for Tuid {
             .map(|tuid| (tuid.nanoseconds_since_epoch(), tuid.inc()))
             .unzip();
 
-        let values = vec![
-            UInt64Array::from_vec(time_ns_values).boxed(),
-            UInt64Array::from_vec(inc_values).boxed(),
+        let values: Vec<ArrayRef> = vec![
+            Arc::new(UInt64Array::from(time_ns_values)),
+            Arc::new(UInt64Array::from(inc_values)),
         ];
         let validity = None;
 
-        let datatype = arrow2::datatypes::DataType::Extension(
-            Self::NAME.to_owned(),
-            Arc::new(Self::arrow2_datatype()),
-            None,
-        );
-
-        // TODO(cmc): We use the extended type here because we rely on it for formatting.
-        Ok(StructArray::new(datatype, values, validity).boxed())
```

Comment:
@teh-cmc this will surely cause a regression, but I don't know where.

Reply:
I can set [`metadata` on the fields](https://docs.rs/arrow/latest/arrow/datatypes/struct.Field.html#method.with_metadata) (which should be equivalent?), but I need some way to test that I'm not breaking anything. I don't know where we rely on this.

Reply:
We do show `ChunkId`:s  the chunk store browser, and everything seems fine there:

<img width="324" alt="Screenshot 2025-01-07 at 20 06 29" src="https://github.com/user-attachments/assets/da7570f2-d77f-48b2-b32d-2c1672673d41" />

Reply:
So yeah, without know what I'm breaking/fixing 🤷

Reply:
Manually adding an `ARROW:extension:name` in the `TransportChunk` glue seems like the right solution yes.

The best way is to test for this is `Command Palette > Print datastore`. Right now your PR doesn't break it (probably because that still goes through arrow2?), so we can see about that later.